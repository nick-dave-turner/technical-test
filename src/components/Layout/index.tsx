import React from "react";
import styled from "styled-components";

import Header from "../Header";
import Footer from "../Footer";

const Container = styled.div`
  height: calc(100vh - 150px);
  margin-top: 80px;
  overflow: hidden;
`;

const Content = styled.div`
  height: 100%;
  padding: 0 5rem;
  overflow-y: auto;
`;

type LayoutProps = {
  children?: React.ReactNode;
};

const Layout: React.FC<LayoutProps> = ({ children }) => {
  return (
    <React.Fragment>
      <Header />
      <Container>
        <Content>{children}</Content>
      </Container>
      <Footer />
    </React.Fragment>
  );
};

export default Layout;
