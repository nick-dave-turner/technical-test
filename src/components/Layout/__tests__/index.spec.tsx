import React from "react";
import { shallow } from "enzyme";

import Layout from "../";

describe("Layout component", () => {
  it("should match snapshot", () => {
    const component = shallow(
      <Layout>
        <h1>Hello World</h1>
      </Layout>
    );
    expect(component).toMatchSnapshot();
  });
});
