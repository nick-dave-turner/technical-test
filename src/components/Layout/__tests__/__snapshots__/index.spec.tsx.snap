// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component should match snapshot 1`] = `
<Fragment>
  <Header />
  <styled.div>
    <styled.div>
      <h1>
        Hello World
      </h1>
    </styled.div>
  </styled.div>
  <Footer />
</Fragment>
`;
